File: api/application/usecase/responsemodel/salon.go
package responsemodel

import "time"

// Salon response構造体
type Salon struct {
	ID           int64     `json:"id"`
	RandID       string    `json:"randId"`
	Name         string    `json:"name"`
	PhoneNumber  string    `json:"phoneNumber"`
	OpeningHours string    `json:"openingHours"`
	ClosingHours string    `json:"closingHours"`
	PostalCode   string    `json:"postalCode"`
	Prefectures  string    `json:"prefectures"`
	City         string    `json:"city"`
	Town         string    `json:"town"`
	AddressCode  string    `json:"addressCode"`
	AddressOther string    `json:"addressOther"`
	CreatedAt    time.Time `json:"createdAt"`
	UpdatedAt    time.Time `json:"updatedAt"`
}

// SalonFind 美容院検索 response構造体
type SalonFind struct {
	Salons []*Salon `json:"salons"`
}

// SalonFindNotBelongs 美容師が所属してない美容院検索
type SalonFindNotBelongs struct {
	Salons []*Salon `json:"salons"`
}

// SalonCreate 美容室作成
type SalonCreate struct {
	*Salon `json:"salon"`
}

// SalonMyPageGet 美容院マイページ取得
type SalonMyPageGet struct {
	*Salon       `json:"salon"`
	Reservations []*Reservation `json:"reservations"`
	Users        []*User        `json:"users"`
}

File: web/src/package/interface/Salon.ts
export interface Salon {
	randId: string
	name: string
	phoneNumber: string
	openingHours: string
	closingHours: string
	postalCode: string
	prefectures: string
	city: string
	town: string
	addressCode: string
	addressOther: string
}

export const initSalon: Salon = <Salon>{}

export const isSalonInterface = (arg: any): arg is Salon => {
  return arg !== null &&
    typeof arg === "object" &&
		typeof arg.randId === "string" && typeof arg.name === "string" && typeof arg.phoneNumber === "string" &&
		typeof arg.openingHours === "string" && typeof arg.closingHours === "string" && typeof arg.postalCode === "string" &&
		typeof arg.prefectures === "string" && typeof arg.city === "string" && typeof arg.town === "string" && typeof arg.addressCode === "string" &&
		typeof arg.addressOther === "string"
}